File: SpiralMatrix.py
#Spiral Matrix

def spiral_matrix(n):
    matrix = [[0]*n for _ in range(n)]
    value = 1
    top, bottom, left, right = 0, n-1, 0, n-1

    while value <= n*n:
        for i in range(left, right+1):
            matrix[top][i] = value
            value += 1
        top += 1

        for i in range(top, bottom+1):
            matrix[i][right] = value
            value += 1
        right -= 1

        for i in range(right, left-1, -1):
            matrix[bottom][i] = value
            value += 1
        bottom -= 1

        for i in range(bottom, top-1, -1):
            matrix[i][left] = value
            value += 1
        left += 1

    return matrix

# 3x3 matrix
mat = spiral_matrix(3)
print("Spiral Matrix:")
for row in mat:
    print("\t".join(map(str, row)))

File: SpiralMatrix.c
//Spiral matrix

#include <stdio.h>

int main() {
    int n = 3,i,j;
    int matrix[3][3] = {0};
    int value = 1;
    int top = 0, bottom = n - 1, left = 0, right = n - 1;

    while (value <= 9) {
        for (i = left; i <= right; i++)
            matrix[top][i] = value++;
        top++;

        for (i = top; i <= bottom; i++)
            matrix[i][right] = value++;
        right--;

        for (i = right; i >= left; i--)
            matrix[bottom][i] = value++;
        bottom--;

        for (i = bottom; i >= top; i--)
            matrix[i][left] = value++;
        left++;
    }

    printf("Spiral Matrix:\n");
    for (i = 0; i < n; i++) {
        for (j = 0; j < n; j++)
            printf("%d\t", matrix[i][j]);
        printf("\n");
    }

    return 0;
}

